homework/src/app.js

```diff
@@ -1,11 +1,254 @@
+
+// Arrays
+
+// 1. Consider the following array:
+
+const favoriteAnimals = ['blowfish', 'capricorn', 'giraffe'];
+
+// Add a statement that adds your favorite animal say 'turtle' to the existing array.
+
+favoriteAnimals.push('turtle');
+
+// Explain in a comment what do you think the new array will look like.
+
+// i think my new array will be ["blowfish", "capricorn", "giraffe", "turtle"]; because push property add the new element at the end of the current array.
+
+// Log your new array.
+
+console.log(favoriteAnimals);
+
+// Now add another favorite animal to the array but make sure it is placed after 'blowfish' and before 'capricorn'.
+favoriteAnimals.splice(1, 0, 'lion');
+
+// Explain in a comment what do you think the new array will look like.
+
+// i think the new array will be : ["blowfish", "lion" "capricorn", "giraffe", "turtle"]
+
+// Log your new array.
+console.log(favoriteAnimals);
+
+// Delete 'giraffe' from the array.
+
+favoriteAnimals.splice(3, 1);
+
+// OPERATORS
+
+// Consider the following snippet:
+
+0 == '0'; // true
+0 == []; // true
+'0' == []; // false
+
+// Mathematically, if a == b and b == c, then a == c. Can you explain in your words why this is not true for the above snippet?
+
+0 == '0'; // true (although their types are not the same their values are equal; thats why its true.)
+0 == []; // true (== coerce empty arrays into its numeric value, which its length, so the length of [] is 0; that's why it is true)
+'0' == []; // false (i guess the object [] is not equal to a string thats why its false.)
+
+// Consider the following snippet:
+
+const x = [1, 2, 3];
+const y = [1, 2, 3];
+const z = y;
+
+// What do you think will happen with x == y, x === y and z === y and z == x?
+
+x == y; // false
+x === y; // false
+z === y; // true,
+z == x; // false,
+
+// Try to think of answer first before running it on browser. Also, explain in your words why this happens?
+
+x == y; // false because their types are objects and they reference different objects
+x === y; // false because as x==y then they can never be x===y
+z === y; // true, because they reference the same objects
+z == x; // false, because they reference different objects
+
+// Consider the following snippet:
+
+const o1 = { foo: 'bar' };
+const o2 = { foo: 'bar' };
+const o3 = o2;
+
+// Show that changing o2 changes o3 (or not) and changing o1 changes o3(or not).
+o2.key = 'changing';
+
+o2; //it would be {foo: 'bar', key: 'changing'}
+o3; // it would also be  {foo: 'bar', key: 'changing'}
+
+// changing o1 does not change o3 because they are not == ;
+
+// Does the order that you assign (o3 = o2 or o2 = o3) matter?
+// Answer: the assign order matters for the above example if we say const o3=o2; it would give us an error because we hav not defined o2 yet.
+
+// THE SECOND PART OF PREVIOUS ASSIGNMENT
+
+// 8.1
+
+let myFavoriteAnimals = ['lions', 'horses', 'eagles'];
+const myFavoriteTennisPlayer = 'Nadal';
+let myAge = 30;
+var answer = false;
```

Comment:
Any reason of using different keywords for creating all these variables instead of just using `const`? 🙂